sdk/provisioning/Azure.Provisioning.CloudMachine/src/Core/ClientWorkspace.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Core;
+
+public abstract class ClientWorkspace
+{
+    public abstract ClientConnectionOptions GetConnectionOptions(Type clientType, string? instanceId = default);
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public ClientCache Subclients { get; } = new ClientCache();
+}
+
+public readonly struct ClientConnectionOptions
+{
+    public ClientConnectionOptions(Uri endpoint, string apiKey)
```

Comment:
Nit - it might be worth changing this to `ApiKeyCred` in the future so you have more wiggle-room with how you evolve the API.  Not blocking now.

Reply:
I was thinking about it, but I wanted it to work with both Azure.Core clients and SCM clients.

Reply:
`AzureKeyCred` already inherits from `ApiKeyCred` so you'll be able to make this work for both if you decide to change it in the future.